src/v/cloud_topics/dl_stm/dl_stm_api.h

```diff
@@ -41,6 +43,18 @@ class dl_stm_api {
     /// available offset.
     std::optional<dl_overlay> lower_bound(kafka::offset offset) const;
 
+    /// Request a new snapshot to be created.
+    ss::future<checked<dl_snapshot_id, dl_stm_api_errc>> start_snapshot();
+
+    /// Read the payload of a snapshot.
+    std::optional<dl_snapshot_payload> read_snapshot(dl_snapshot_id id);
+
+    /// Remove all snapshots with version less than the given version.
+    /// This must be called periodically as new snapshots are being created
+    /// to avoid the state growing indefinitely.
+    ss::future<checked<void, dl_stm_api_errc>>
+    remove_snapshots_before(dl_version last_version_to_keep);
+
```

Comment:
this class needs a gate and a shutdown code path if the object is not transient (only created to replicate one message)